### DISCUSSION TITLE
Support for Dependent Fields on Action

### DISCUSSION THREAD

**Discussion by kevariable** [laravel/nova-issues#3931] [Ideas & Feature Requests]
- Laravel Version: ^9.0
- Nova Version: "^4.0"
- PHP Version: "php": "^8.0",
- Database Driver & Version: MySQL version 8
- Operating System and Version: Arch Linux
- Browser type and version: Chromium Latest Version
- Reproduction Repository: -

### Description:

Depends On feature doesn't work after upgrading from Laravel Nova 3 for Action

### Detailed steps to reproduce the issue on a fresh Nova installation:

Following all upgrade guide

## Code

```php
Select::make('Name', 'name')
      ->options([
          'xxx' => 'xxx'
      ])->rules('required'),

  Text::make('Store Name')
       ->hide()
      ->dependsOn('name', function (Text $field, NovaRequest $request, FormData $formData) {
      
         // I tries to Log no data was attached here
      
          $field
             ->show()
              ->rules([
                  'required',
                  'max:15',
                  Rule::unique(Channel::class)->where('name', $formData->name)
              ])
              ->help(Str::replace(':name',  $formData->name, 'Please enter your :name xxx name'));
      }),
```

## Screenshots

![image](https://user-images.githubusercontent.com/56016477/162344362-27a274ff-e413-45ee-a8f0-0861ceb14b06.png)

Let me know if need more details on specify issue

**Comment by crynobone** [comment]
Action doesn't implement depends on at the moment.

**Reply by dkulyk** [reply]
@crynobone This [patch](https://gist.github.com/dkulyk/6a02a1f7bd592dfd03ddd4ab81aef8e4) will added this functional to Nova 4. 
I used Nova 4.5.4 as base.

**Reply by Grafikart** [reply]
Thanks for the patch. I would say this method seem to skip some parameters when sending the patch request (viaResource, viaResourceId, viaRelationship). I added this to the patch request but it's 

```
{
  editing: true,
  editMode: this.editMode,
  viaResource: this.viaResource || this?.$parent?.$parent.resourceName,
  viaResourceId: this.viaResourceId || this?.$parent?.$parent?.selectedResources[0],
  resourceId: this.viaResourceId || this?.$parent?.$parent?.selectedResources[0],
  viaRelationship: this.viaRelationship,
  field: this.field.attribute,
  component: this.field.component,
}
```

**Reply by crynobone** [reply]
@Grafikart can you share an example where this can a problem as simple as possible?

**Reply by Grafikart** [reply]
@crynobone For instance we have an action on a User resource and we use the resourceID 

```php
public function fields(NovaRequest $request): array
    $userId = $request->get('resourceId');
    $user = User::find($userId);
    $adresses = $user->addresses()->pluck('address', 'id'); // This is used for the select field

    return [
        // .....
    ];
}
```

With the patch, since we don't receive the resourceId information on actions we are not able to filter addresses for the user (the action is called on the user detail page).

**Reply by Francois-Francois** [reply]
@crynobone Do you have any update on this feature ? This is locking many users from upgrading to Nova 4 :/

**Comment by kevariable** [comment]
Hi, thanks to up this for discussion i will appreciate it. Currently our production site with many merchants rely on Action, it will be nice if depends on available on that

**Reply by infinitypaul** [reply]
Same here

**Comment by hivokas** [comment]
I like this idea, I also need this.

**Comment by elizdlt202** [comment]
same here! 
-> dependsOn is needed in actions for my project too

Please and many thanks!

**Comment by meyer59** [comment]
Needed for Action Too 🙌

**Comment by mikkellindblom** [comment]
Would love this feature 🥇

**Comment by vilhelmjosander** [comment]
Same here!

**Comment by andrefedalto** [comment]
Same here. Currently assessing if an existing project can be upgraded to Nova 4, but the existing package we use (https://github.com/epartment/nova-dependency-container) does not work with Nova 4 and the core functionality that was implemented does not work within Actions. So this became a blocker for our upgrade.

**Comment by crynobone** [comment]
Thank to @dkulyk (https://github.com/laravel/nova-issues/discussions/3931#discussioncomment-2741417) we making good progress with this feature.

**Reply by dkulyk** [reply]
I think better is pass a custom patch link to field instead  `action` property
This will allow to use fields patching on external packages like [whitecube/nova-flexible-content](https://github.com/whitecube/nova-flexible-content)
I updated the patch for this

**Reply by crynobone** [reply]
already made those changes to support inline action on table row.

**Comment by sutcliffe121** [comment]
It would be great to have this feature, currently blocking my users from using the upgrade from Nova 3 to 4.

**Comment by crynobone** [comment]
PR has been made and is pending review.

**Comment by Francois-Francois** [comment]
@crynobone do you have ETA for this release ?

**Reply by crynobone** [reply]
See https://github.com/laravel/nova-issues/discussions/3931#discussioncomment-2861743

**Comment by Francois-Francois** [comment]
@crynobone I'm asking for the release date, now PR has been in review since many days.

**Reply by crynobone** [reply]
A complex PR will requires days to weeks before being merged. This PR requires new API and frontend changes that need to be approved therefore it requires more times. 

Once the changes is accepted then we can estimate the release date.

**Comment by Francois-Francois** [comment]
@crynobone Still no news about this feature, although there have been 2 releases.

The community has written some of the necessary code. Big projects are blocked in their migration because of this regression, and finally, the licenses for nova4 are not usable because of this (especially since the licenses are annual now).

What can we do to get you to focus on this request, so that we can finally benefit from this update ?

Happy to help if needed ! :)

**Comment by seche** [comment]
We could use this feature as well. Any news on this?
